core/tauri/src/manager/mod.rs

```diff
@@ -0,0 +1,792 @@
+          }
+        }
+
+        acc
+      });
+
+  let dangerous_disable_asset_csp_modification = &manager
+    .config()
+    .tauri
+    .security
+    .dangerous_disable_asset_csp_modification;
+  if dangerous_disable_asset_csp_modification.can_modify("script-src") {
+    replace_csp_nonce(
+      asset,
+      SCRIPT_NONCE_TOKEN,
+      &mut csp,
+      "script-src",
+      hash_strings.script,
+    );
+  }
+
+  if dangerous_disable_asset_csp_modification.can_modify("style-src") {
+    replace_csp_nonce(
+      asset,
+      STYLE_NONCE_TOKEN,
+      &mut csp,
+      "style-src",
+      hash_strings.style,
+    );
+  }
+
+  #[cfg(feature = "isolation")]
+  if let Pattern::Isolation { schema, .. } = &*manager.pattern {
+    let default_src = csp
+      .entry("default-src".into())
+      .or_insert_with(Default::default);
+    default_src.push(crate::pattern::format_real_schema(schema));
+  }
+
+  Csp::DirectiveMap(csp).to_string()
+}
+
+// inspired by https://github.com/rust-lang/rust/blob/1be5c8f90912c446ecbdc405cbc4a89f9acd20fd/library/alloc/src/str.rs#L260-L297
+fn replace_with_callback<F: FnMut() -> String>(
+  original: &str,
+  pattern: &str,
+  mut replacement: F,
+) -> String {
+  let mut result = String::new();
+  let mut last_end = 0;
+  for (start, part) in original.match_indices(pattern) {
+    result.push_str(unsafe { original.get_unchecked(last_end..start) });
+    result.push_str(&replacement());
+    last_end = start + part.len();
+  }
+  result.push_str(unsafe { original.get_unchecked(last_end..original.len()) });
+  result
+}
+
+fn replace_csp_nonce(
+  asset: &mut String,
+  token: &str,
+  csp: &mut HashMap<String, CspDirectiveSources>,
+  directive: &str,
+  hashes: Vec<String>,
+) {
+  let mut nonces = Vec::new();
+  *asset = replace_with_callback(asset, token, || {
+    #[cfg(target_pointer_width = "64")]
+    let mut raw = [0u8; 8];
+    #[cfg(target_pointer_width = "32")]
+    let mut raw = [0u8; 4];
+    #[cfg(target_pointer_width = "16")]
+    let mut raw = [0u8; 2];
+    getrandom::getrandom(&mut raw).expect("failed to get random bytes");
+    let nonce = usize::from_ne_bytes(raw);
+    nonces.push(nonce);
+    nonce.to_string()
+  });
+
+  if !(nonces.is_empty() && hashes.is_empty()) {
+    let nonce_sources = nonces
+      .into_iter()
+      .map(|n| format!("'nonce-{n}'"))
+      .collect::<Vec<String>>();
+    let sources = csp.entry(directive.into()).or_default();
+    let self_source = "'self'".to_string();
+    if !sources.contains(&self_source) {
+      sources.push(self_source);
+    }
+    sources.extend(nonce_sources);
+    sources.extend(hashes);
+  }
+}
+
+/// A resolved asset.
+pub struct Asset {
+  /// The asset bytes.
+  pub bytes: Vec<u8>,
+  /// The asset's mime type.
+  pub mime_type: String,
+  /// The `Content-Security-Policy` header value.
+  pub csp_header: Option<String>,
+}
+
+#[default_runtime(crate::Wry, wry)]
+pub struct AppManager<R: Runtime> {
```

Comment:
should probably also move this to `manager/app.rs`

Reply:
i'm not really sure, since AppManager is the root manager it can live in mod.rs IMO

Reply:
Yeah I agree but app.rs gives more clarity as it is not clear from first glance where AppManager lives.

Reply:
anyways, this is just a nit so go with your gut.